core/camel-core-engine/src/main/java/org/apache/camel/impl/DefaultCamelContext.java

```diff
@@ -975,6 +991,32 @@ protected synchronized boolean removeRoute(String routeId, LoggingLevel loggingL
         return removed;
     }
 
+    /**
+     * Indicates whether the route should be included according to the precondition.
+     *
+     * @param  definition the definition of the route to check.
+     * @return            {@code true} if the route should be included, {@code false} otherwise.
+     */
+    private boolean includedRoute(RouteDefinition definition) {
+        final String precondition = definition.getPrecondition();
+        if (precondition == null) {
+            LOG.debug("No precondition found, the route is included by default");
```

Comment:
I think trace level is fine here, as otherwise debug is a bit spammed as 99,9% of routes are included